radix-engine-tests/tests/db/substate_database_overlay.rs

```diff
@@ -484,30 +484,59 @@ fn run_scenarios(
         .build();
 
     let mut next_nonce: u32 = 0;
-    for scenario_builder in get_builder_for_every_scenario() {
-        let epoch = ledger1.get_current_epoch();
-        let mut scenario = scenario_builder(ScenarioCore::new(network.clone(), epoch, next_nonce));
-        let mut previous = None;
-        loop {
-            let next = scenario
-                .next(previous.as_ref())
-                .map_err(|err| err.into_full(&scenario))
-                .unwrap();
-            match next {
-                NextAction::Transaction(next) => {
-                    let receipt1 = ledger1.execute_notarized_transaction(&next.raw_transaction);
-                    let receipt2 = ledger2.execute_notarized_transaction(&next.raw_transaction);
-
-                    check_callback(
-                        (ledger1.substate_db(), &receipt1),
-                        (ledger2.substate_db(), &receipt2),
-                    );
-
-                    previous = Some(receipt1);
-                }
-                NextAction::Completed(end_state) => {
-                    next_nonce = end_state.next_unused_nonce;
-                    break;
+    let mut scenario_builders = scenario_builders();
+    for requirement in ScenarioRequirements::all() {
+        if let ScenarioRequirements::ProtocolUpdateUpTo(protocol_update) = requirement {
+            protocol_update
+                .generate_state_updates(ledger1.substate_db_mut(), &network)
+                .into_iter()
+                .for_each(|state_updates| {
+                    ledger1
+                        .substate_db_mut()
+                        .commit(&state_updates.create_database_updates::<SpreadPrefixKeyMapper>())
+                });
+            protocol_update
+                .generate_state_updates(ledger2.substate_db_mut(), &network)
+                .into_iter()
+                .for_each(|state_updates| {
+                    ledger2
+                        .substate_db_mut()
+                        .commit(&state_updates.create_database_updates::<SpreadPrefixKeyMapper>())
+                })
+        };
+
+        let scenario_builders = scenario_builders
+            .shift_remove(requirement)
+            .unwrap_or_default();
```

Comment:
Hey! We're not making that assumption, the requirements can be specified in any order and they will be executed in Genesis -> Anemone -> Bottlenose order regardless of how they're specified. This is because the first for loop in this function iterates over `ScenarioRequirements::all()` which defines the canonical order of the execution of scenarios.

How do you feel about the above? We can also specify them in a vector here an make the order more explicit and clear to clients of the executor.

Reply:
Commenting an update here, I've refactored the implementation quite a bit based on David's feedback, the protocol requirement of the scenarios is now specified on the `ScenarioCreator` of the scenario and the order of the scenarios as a whole is defined in the executor.